<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="This package is designed to work with Human Surfaceome genes/proteins. One of 
  the first uses is to categorize what are surfaceome genes (there are several sources). 
  Additionally, the goal is to include tools for identifying the expressed surfaceome in
  a molecular experiment.">
<title>Human Surfaceome Annotation • surfaceome</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Human Surfaceome Annotation">
<meta property="og:description" content="This package is designed to work with Human Surfaceome genes/proteins. One of 
  the first uses is to categorize what are surfaceome genes (there are several sources). 
  Additionally, the goal is to include tools for identifying the expressed surfaceome in
  a molecular experiment.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">surfaceome</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/surfaceome_definitions.html">Surfaceome Definitions</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="surfaceome">Surfaceome<a class="anchor" aria-label="anchor" href="#surfaceome"></a>
</h1></div>
<p>There is a subset of the genome (or proteome) that lives at the cell surface. This includes things like receptors, but also includes other types of proteins. This package uses source data from publications to define the surfaceome and implements some tools for working with it.</p>
<p>Note: Most of this package is at present aspirational other than the data.</p>
<div class="section level2">
<h2 id="source-data">Source Data<a class="anchor" aria-label="anchor" href="#source-data"></a>
</h2>
<p>There are two sources of surfaceome annotation, CSPA and Surfy. See below for details on each of these. This data will be available via data frames within the package.</p>
<div class="section level3">
<h3 id="cspa">CSPA<a class="anchor" aria-label="anchor" href="#cspa"></a>
</h3>
<p>The Cell Surface Protein Atlas (CSPA, <a href="https://wlab.ethz.ch/cspa/" class="external-link uri">https://wlab.ethz.ch/cspa/</a>) profiled a bunch of cells using mass spec to empirically determine what proteins are on the cell surface.</p>
<p>These proteins are annotated in: <a href="https://wlab.ethz.ch/cspa/data/S2_File.xlsx" class="external-link uri">https://wlab.ethz.ch/cspa/data/S2_File.xlsx</a></p>
<p>Note: It appears that the S2_File.xlsx is identical to pone.0121314.s003.xlsx from the publication. The file itself consists of three different sheets. From the publication:</p>
<pre><code>S2 File: CSPA validated surfaceome proteins. Excel file containing all human and mouse surfaceome proteins in two tables and an additional table with all identified N-glycopeptides. A. List of 1492 human surfaceome proteins and their annotation. B. List of 1296 mouse surfaceome proteins and their annotation. C. List of 13942 mouse and human derived N-glycopeptides, including identified modified form.</code></pre>
<p>The sheet of interest is Sheet A (called ‘Table A’ in the spreadsheet).</p>
</div>
<div class="section level3">
<h3 id="surfy">SURFY<a class="anchor" aria-label="anchor" href="#surfy"></a>
</h3>
<p>This group also used machine learning to try and predict all of the cell surface proteins in the proteome, training from the CSPA. It used information on sequence and domains to do these predictions.</p>
<p>These proteins are annotated in: <a href="http://wlab.ethz.ch/surfaceome/table_S3_surfaceome.xlsx" class="external-link uri">http://wlab.ethz.ch/surfaceome/table_S3_surfaceome.xlsx</a></p>
<p>The “in silico surfaceome only” tab has the information (source can be machine learning, pos. trainingset or GPI (Uniprot). The total is 2,886 proteins. There is a ‘UniProt gene’ column with a gene name. There is also a column ‘GeneID’ which might be more robust to changes in names (if it’s present).</p>
</div>
</div>
<div class="section level2">
<h2 id="summarization">Summarization<a class="anchor" aria-label="anchor" href="#summarization"></a>
</h2>
<p>Although you can use the individual gene/protein measurements as individual members of the surfaceome, you can also summarize these to a single measure. Note this is not always desireable, as the difference in specific markers is what is most important. However, if you are interested in summarizing there are two approaches:</p>
<ul>
<li>PCA</li>
<li>ssgsea</li>
</ul>
<p>With this, we can score the surfaceome overall or provide individual measures by different categories.</p>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>This is not yet implemented, however the goal of this part of the package is to provide visualization functions for gene expression and/or protein data. Look at it like immunophenogram, heatmap of samples for surfaceome but separated out by category (assignment) or class.</p>
<p>Note: from the literature I have seen PCA and heatmap visualizations of the surfaceome hence the thought that this would be appropriate.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing surfaceome</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Steven Eschrich <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/YOUR-ORCID-ID" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Steven Eschrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
